model/project_ref.go

```diff
@@ -405,28 +406,26 @@ func FindMergedProjectRef(identifier string) (*ProjectRef, error) {
 }
 
 func mergeBranchAndRepoSettings(pRef *ProjectRef, repoRef *RepoRef) *ProjectRef {
-	// if a setting is disabled overall, then disable the branch
-	pRef.Owner = repoRef.Owner
-	pRef.Repo = repoRef.Repo
-	if !repoRef.Enabled {
-		pRef.Enabled = repoRef.Enabled
-	}
-	if repoRef.PatchingDisabled {
-		pRef.PatchingDisabled = repoRef.PatchingDisabled
-	}
-	if repoRef.RepotrackerDisabled {
-		pRef.RepotrackerDisabled = repoRef.RepotrackerDisabled
-	}
-	if pRef.DispatchingDisabled {
-		pRef.DispatchingDisabled = repoRef.DispatchingDisabled
-	}
-	// if the following fields aren't configured, default to the repo configuration
-	if pRef.RemotePath == "" {
-		pRef.RemotePath = repoRef.RemotePath
+	// if the setting is not defined in the project, default to the repo settings.
+	// For booleans, we default to the repo setting if the boolean is false, except for enabled.
+	reflectedBranch := reflect.ValueOf(pRef)
+	reflectedRepo := reflect.ValueOf(repoRef)
+
+	isProjectDisabled := !pRef.Enabled
+	for i := 0; i < reflectedBranch.Elem().NumField(); i++ {
+		branchField := reflectedBranch.Elem().FieldByIndex([]int{i})
```

Comment:
I think you can do reflectedBranch.Field(i) to get the same thing here but cleaner (same with below)